base/path.jl

```diff
@@ -39,7 +43,12 @@ end
         m = match(r"^(\w+:|\\\\\w+\\\w+|\\\\\?\\UNC\\\w+\\\w+|\\\\\?\\\w+:|)(.*)$", path)
         bytestring(m.captures[1]), bytestring(m.captures[2])
     end
-    homedir() = get(ENV,"HOME",string(ENV["HOMEDRIVE"],ENV["HOMEPATH"]))
+    function homedir()
+        haskey(ENV, "HOME") ||
+            (haskey(ENV, "HOMEDRIVE") && haskey(ENV["HOMEPATH"])) ||
```

Comment:
`haskey(ENV, "HOMEPATH")`

also inconsistent indentation below on the `get` line